lib/jackalope/work_list.ex

```diff
@@ -42,6 +43,35 @@ defmodule Jackalope.WorkList do
     %__MODULE__{work_list | items: tl(work_list.items)}
   end
 
+  @spec pending(t(), reference()) :: t()
+  def pending(work_list, ref) do
+    item = hd(work_list.items)
+
+    %__MODULE__{
+      work_list
+      | items: tl(work_list.items),
+        pending: Map.put(work_list.pending, ref, item)
+    }
+  end
+
+  @spec reset_pending(Jackalope.WorkList.t()) :: Jackalope.WorkList.t()
+  def reset_pending(work_list) do
+    pending_items =
+      for(
+        {ref, item} when is_reference(ref) <- work_list.pending,
+        do: item
+      )
```

Comment:
Is this just the following:

```suggestion
    pending_items = Map.values(pending)
```